tests/test_replies.js

```diff
@@ -262,6 +262,50 @@ test('headers work', async t => {
   scope.done()
 })
 
+test('raw headers work with reply callback but no status', async t => {
+  const scope = nock('http://example.com')
+    .get('/')
+    .reply(() => 'OK!', { 'X-My-Headers': 'My Header value' })
+
+  const { headers, rawHeaders, statusCode } = await got('http://example.com/')
+
+  t.equal(statusCode, 200)
+  t.equivalent(headers, { 'x-my-headers': 'My Header value' })
+  t.equivalent(rawHeaders, ['X-My-Headers', 'My Header value'])
+  scope.done()
+})
+
+test('raw headers merge with reply callbacks array with headers', async t => {
+  const scope = nock('http://example.test')
+    .get('/')
+    .reply(
+      () => [202, 'body', { 'x-key': 'value', 'x-key-2': 'value 2 override' }],
+      { 'x-key-2': 'value 2', 'x-key-3': 'value 3' }
+    )
```

Comment:
Oh no! I didn't realize it was possible to pass the headers to the callback. That reverses what I wrote at https://github.com/nock/nock/issues/1513#issuecomment-488320407. I'm sorry for my misunderstanding!

I do not think we need to provide so many different ways of doing this. 😝 

What about supporting these forms:

1. `.reply(() => 'body')`
2. `.reply(() => ['body', { 'x-key': 'value', 'x-key-2': 'value 2 override' }])`
3. `.reply(() => [202, 'body', { 'x-key': 'value', 'x-key-2': 'value 2 override' }])`
4. `.reply((uri, requestBody, cb) => cb(null, 'body'), 1000))`
4. `.reply((uri, requestBody, cb) => cb(null, ['body', { 'x-key': 'value', 'x-key-2': 'value 2 override' }]), 1000))`
5. `.reply((uri, requestBody, cb) => cb(null, [201, 'body', { 'x-key': 'value', 'x-key-2': 'value 2 override' }]), 1000))`

That way the function and async forms have effectively the same interfaces as the synchronous calls to `.reply()`.

Ideally all these paths would be tested, and the documentation would make it clear what is possible.

Thoughts?

Reply:
I don't think my issue/question/change relates to the sync vs async forms of the callback.

I thinks it's more about a confusion behind the intended signature of the method, regardless of what the callbadk returns..
My original assumption was the intended signature was the two forms below. This seems to closely match how I interpret the docs as they stand today.
`.replay(statusCode, [bodyOrCallback, [rawHeaders]])`
`.replay(callback)`

If that is correct, I think we should update the [beginning of the method](https://github.com/nock/nock/blob/6d832020031409abb3f06d966390158a387beeff/lib/interceptor.js#L80-L84) to the following
```js
  if (_.isFunction(statusCode)) {
    body = statusCode
    statusCode = 200
    rawHeaders = undefined
  }
```
The `arguments.length <= 2` is confusing/misleading, and clearing the `rawHeaders` clarifies the method's intention to anyone digging around, plus reduces the changes for bugs in the future.

That being said, this PR does **not** do this. 
Instead it makes the signature of the method more like this
`.replay([statusCode], [bodyOrCallback, [rawHeaders]])`
or maybe it's better described as
`.replay(statusCode, [bodyOrCallback, [rawHeaders]])`
`.replay(callback, [rawHeaders])`

Personally, I'm less of a fan of having all these ways of accomplishing the same thing, so I'd be happy to close this PR. I just want to make sure we're on the same page.

Reply:
> My original assumption was the intended signature was the two forms below. This seems to closely match how I interpret the docs as they stand today.
> `.replay(statusCode, [bodyOrCallback, [rawHeaders]])`
> `.replay(callback)`

Yes, I agree this is a better interface.

> Personally, I'm less of a fan of having all these ways of accomplishing the same thing, so I'd be happy to close this PR. I just want to make sure we're on the same page.

Sounds good. Thanks!